### PR TITLE
Small improvements for DFT analysis

### PR DESCRIPTION
**Pull request by volkm**
- Instantiator for parametric DFT
- Use less information in DFT exploration heuristics
- Improvements in DFTModularizer
- Initialize relevant events in DFT constructor

### CODE REVIEW

File: src/storm-dft/storage/dft/DFT.cpp [hunk 1]
```diff
@@ -90,6 +90,9 @@ namespace storm {
             //Reserve space for failed spares
             ++mMaxSpareChildCount;
             mStateVectorSize = DFTStateGenerationInfo::getStateVectorSize(nrElements(), mNrOfSpares, mNrRepresentatives, mMaxSpareChildCount);
+
+            // Set TLE as relevant
```
Comment (sjunges) [inline src/storm-dft/storage/dft/DFT.cpp:94]: I find this hard to understand. Maybe extend the comment

File: src/storm-dft/storage/dft/DFT.cpp [hunk 2]
```diff
@@ -1176,6 +1187,45 @@ namespace storm {
             stream << "=========================================\n";
         }
 
+#ifdef STORM_HAVE_CARL
```
Comment (sjunges) [inline src/storm-dft/storage/dft/DFT.cpp:1190]: Do we still use STORM_HAVE_CARL?

File: src/storm-dft/transformations/DftInstantiator.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+namespace storm {
+    namespace transformations {
+        namespace dft {
+
+            template<typename ParametricType, typename ConstantType>
+            DftInstantiator<ParametricType, ConstantType>::DftInstantiator(storm::storage::DFT<ParametricType> const &dft) : dft(dft) {
+            }
+
+            template<typename ParametricType, typename ConstantType>
+            DftInstantiator<ParametricType, ConstantType>::~DftInstantiator() {
```
Comment (sjunges) [inline src/storm-dft/transformations/DftInstantiator.cpp:15]: I think destructors should always be made virtual?

**Comment by sjunges** [comment]
No major concerns. LGTM, you know the DFT code best ;-)

File: src/storm-dft/storage/dft/DFT.cpp (hunk 2 above)
Comment (volkm) [reply src/storm-dft/storage/dft/DFT.cpp]: We still use it in a lot of places. But this a future issue. I removed it here.

File: src/storm-dft/transformations/DftInstantiator.cpp (hunk 3 above)
Comment (volkm) [reply src/storm-dft/transformations/DftInstantiator.cpp]: The destructor is virtual in the header file. I changed it to `= default` in the header file.

**Comment by sjunges** [comment]
I would merge once checks have passed?

**Comment by volkm** [comment]
Sounds good to me.

**Pull request merged**